packages/app-desktop/gui/ConfigScreen/controls/plugins/useOnInstallHandler.test.ts

```diff
@@ -0,0 +1,97 @@
+import useOnInstallHandler from './useOnInstallHandler';
+import { renderHook } from '@testing-library/react-hooks';
+
+import PluginService, { defaultPluginSetting, PluginSetting } from '@joplin/lib/services/plugins/PluginService';
+import { ItemEvent } from './PluginBox';
+import produce from 'immer';
+
+jest.mock('@joplin/lib/services/plugins/PluginService');
+jest.mock('immer');
+
+const pluginServiceinstance = {
+	updatePluginFromRepo: jest.fn(),
+	installPluginFromRepo: jest.fn(),
+};
+
+const pluginId = 'test.plugin';
+const setInstallingPluginIds = jest.fn();
+const repoApi = jest.fn();
+const onPluginSettingsChange = jest.fn();
+const itemEvent = ({
+	item: { manifest: { id: pluginId } },
+} as ItemEvent);
+const callHook = (isUpdate: boolean, pluginEnabled = true) => () => useOnInstallHandler(
+	setInstallingPluginIds,
+	{
+		[pluginId]: {
+			enabled: pluginEnabled,
+			deleted: false,
+			hasBeenUpdated: false,
+		},
+	},
+	repoApi,
+	onPluginSettingsChange,
+	isUpdate
+);
+
+describe('useOnInstallHandler', () => {
+
+	beforeAll(() => {
+		(PluginService.instance as jest.Mock).mockReturnValue(pluginServiceinstance);
+		(defaultPluginSetting as jest.Mock).mockImplementation(
+			jest.requireActual('@joplin/lib/services/plugins/PluginService').defaultPluginSetting
+		);
+	});
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	test('should set updatingPluginIds while updating', async () => {
+		const { result: { current: cb } } = renderHook(callHook(true));
+		await cb(itemEvent);
```

Comment:
Could you rename "cb" to either "onInstall" or "onUpdate", for consistency with the main code? This applies to all the tests.